### ISSUE TITLE
Fido is implemented in C#

### ISSUE THREAD

**Issue by b** [Netflix/Fido#2]

**Comment by manasdk** [comment]
Yeah. I agree this seems like a major issue.

**Comment by Burnett01** [comment]
Why?

**Comment by b** [comment]
It dramatically reduces the set of people who can contribute, adds more dependencies for non-Windows environments (most of the online service world) and looks nothing like the rest of the NetflixOSS stack.

**Comment by ghuntley** [comment]
@b please go back to /b/

![screen shot 2015-05-05 at 12 32 01 pm](https://cloud.githubusercontent.com/assets/127353/7465011/a300f88c-f30c-11e4-9629-ff6606308272.png)

@netflix thank-you for your contribution to the oss community.

**Comment by b** [comment]
@ghuntley your data and your attitude support my statements. "get out" is not an appropriate response to a difference of opinion, especially when this isn't your house. have a nice day.

**Comment by DanielGilbert** [comment]
Basically - no. If you can write code in Java, than you can write code in C#, too. As for the dependencies: For Java I need a runtime, for C# I need a runtime. No big deal here. So - yes, It's not an issue at all, at least from my perspective. People should use the language they like and that suits their needs. But hey, that's just my humble opinion.

Disclaimer: I'm a C# dev. And I don't like Java. Then again - it's just my personal opinion.

**Comment by matt40k** [comment]
Am I missing something - reading documentation it looks like a way to determine if the workstation needs to be reinstalled. As it looks like they are referring to a windows machine mostly, which includes .net framework by default, wouldn't it make sense to use C#? This isn't their core business, they aren't selling security software, which is why they have OSS it. They want a quick solution to solve a problem, so a language with a large library for Windows functions, ie .NET makes sense surely? How is this an issue? Surely you use the best tool for the job, you don't give a 4 year old a Java IDE and you wouldn't give a senior developer Scratch to write the next Facebook. If you want to rewrite it, then fork it and rewrite it. I don't see how raising this as an issue will help solve a problem. It isn't broken.

Also don't forget Visual Studio \ C# works on Macs and Linux machines now - https://code.visualstudio.com/#alt-downloads

**Comment by TheAngryByrd** [comment]
Yeah this should be F#!:trollface:

**Comment by RyPeck** [comment]
Has anyone at Netflix gotten this built on a Linux distribution? Might be good to have a Vagrantfile that gives everyone an environment to get the basics up and going.

**Comment by robfry** [comment]
Not to stop the debate, but I thought I'd chime in. I do appreciate everyone providing feedback and opinions. I'm pretty agnostic when it comes to technology and don't participate in the debates about why one language or technology was used over another. To me it's like debating the value of a hammer versus the value of a wrench. :) Respect and understands everyone's opinions and I hope everyone selects whatever tool they feel fits the job.

That said, the reason FIDO is in C# was more one of circumstance. The original version was a 100-ish lines of bash script. After the initial success I started the second version in Java, but at the time I was wanting to pick up another language. In parallel Oracle was going through a rough patch with security around Java which annoyed me and C# was a natural pathway where the learning curve would be mostly small and I could reuse some code. In roughly early 2012-ish, Netflix had not created quite the splash in the OSS community, so I was writing FIDO in a very proprietary way. I never thought it would see the light of day. :) By the time someone came to me to ask me to open source FIDO and I was already too far in to change path. I'm an architect at Netflix, not a full-time developer, and what you see in FIDO has been done in about 15%-20% of my daily work. If you consider the success we were having with FIDO, plus the enhancements/features needed, and additional integrations constantly being built in addition to my normal day job... re-writing FIDO in something else was never a serious option for me.

Anyhow, hope that clears up the 'why C#' questions.

**Comment by Burnett01** [comment]
Thanks to @b and @robfry for clearification

**Comment by IDisposable** [comment]
Can we close this issue, I suggest the label wont-fix

**Comment by robfry** [comment]
Not a bug, won't fix.

**Comment by harryxy12345** [comment]
Please ROB thank, you close it!!!
